File: DailyLeetcode/9Aug/PowerOfTwo.cpp
//Bit manipulation
//Time Complexity:O(1)
//Space Complexity:O(1)
class Solution {
public:
    bool isPowerOfTwo(int n) {
        if(n<=0){
            return false;
        }
        return (n&(n-1))==0;
    }
};

//Recursion
//Time Complexity:O(logn)
//Space Complexity:O(1)
class Solution {
public:
    bool isPowerOfTwo(int n) {
        if(n==0 || n==1){
            return n;
        }
        if(n%2!=0){
            return false;
        }
        return isPowerOfTwo(n/2);
    }
};

File: DailyLeetcode/9Aug/PowerOfTwo.py
# TIme-Complexity-O(n)
# Space Complexity-O(1)
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n<1:
            return False

        x=1
        while x<n:
            x=x*2
        return x==n
        